### ISSUE TITLE
Constrain connector resource usage

### ISSUE THREAD

**Issue by jshearer** [estuary/flow#660]
A few days ago we had a pretty bad experience trying to get a customer onboard, ultimately caused by the Firestore connector having "unfriendly" memory usage patterns that scale linearly with the size of the dataset being captured. Due to the fact that the flow reactor runs multiple workloads (captures, derivations, materializations) per instance (so, per pod in this case), not only did this specific firestore capture run out of memory, it also brought down all of the other workloads running on whatever flow-reactor instance it was scheduled on due to the entire pod getting oom-killed.

While we should absolutely fix the firestore capture connector to not use so much memory, the fact that one noisy neighbor can impact other unrelated workloads like this is an issue to address.

* **Captures** and **materializations** run as Docker containers run on the `dind` container in the `flow-reactor` pod. Conveniently, these are the easiest to constrain and also the most likely to cause resource exhaustion (is this true?). All we need to do is ask Docker to impose [runtime resource constraints](https://docs.docker.com/config/containers/resource_constraints/)
* **Derivations** run inside the `flow-reactor` main container as separate NodeJS processes, interacted with over a local HTTP api. We could either attempt to constrain these nodejs processes directly via cgroups, or we could move to running these as Docker containers. Given the discussions we're having about eventually running (everything? just derivations?) in Firecracker containers, it might be worth not putting in the effort to change how we run derivations until it's time to do that work.

Almost as important as constraining resource usage to prevent noisy neighbors, at least in my opinion, is monitoring these workloads to see if they actually are reaching their limits, and if they are, alerting about it like we do today with k8s OOM-kills. We already have metrics in Grafana... **should we just add cpu and memory metrics for each task**?

So TL;DR: we should add `--cpus="1" --memory="1g"` (or something) to our invocation of captures and materializations, and possibly monitoring for cpu/memory of captures and materializations

**Comment by dhammika** [comment]
At the moment, `dind` is a bit of a black box. It's mem usage is fluctuating quite a bit. I'm not sure if that's from new captures or some sort of backfill jobs. 
![dind-mem](https://user-images.githubusercontent.com/262792/197898655-282fe583-0c24-4f30-8aec-60a0b7adae47.png)
